include/ddc/kernels/splines/bsplines_non_uniform.hpp

```diff
@@ -70,125 +95,301 @@ class NonUniformBSplines : detail::NonUniformBSplinesBase
+         *
+         * @param x A reference to another Impl
+         * @return A reference to the copy Impl
+         */
         Impl& operator=(Impl const& x) = default;
 
+        /** @brief Move-assigns
+         *
+         * @param x An rvalue to another Impl
+         * @return A reference to the moved Impl
+         */
         Impl& operator=(Impl&& x) = default;
 
+        /** @brief Evaluates non-zero B-splines at a given coordinate.
+         *
+         * The values are computed for every B-spline with support at the given coordinate x. There are only (degree+1)
+         * B-splines which are non-zero at any given point. It is these B-splines which are evaluated.
+         * This can be useful to calculate a spline approximation of a function. A spline approximation at coordinate x
+         * is a linear combination of these B-spline evaluations weighted with spline coefficients of the spline-transformed
+         * initial discrete function.
+         *
+         * @param[out] values The values of the B-splines evaluated at coordinate x. It has to be a 1D mdspan with (degree+1) elements.
+         * @param[in] x The coordinate where B-splines are evaluated.
+         * @return The index of the first B-spline which is evaluated.
+         */
         KOKKOS_INLINE_FUNCTION discrete_element_type
         eval_basis(DSpan1D values, ddc::Coordinate<Tag> const& x) const;
 
+        /** @brief Evaluates non-zero B-splines derivatives at a given coordinate
+         *
+         * The derivatives are computed for every B-splines with support at the given coordinate x. There are only (degree+1)
+         * B-splines which are non-zero at any given point. It is these B-splines which are derivated.
+         * A spline approximation of a derivative at coordinate x is a linear
+         * combination of those B-splines derivatives weigthed with splines coefficients of the spline-transformed
+         * initial discrete function.
+         *
+         * @param[out] derivs The derivatives of the B-splines evaluated at coordinate x. It has to be a 1D mdspan with (degree+1) elements.
+         * @param[in] x The coordinate where B-splines derivatives are evaluated.
+         * @return The index of the first B-spline which is derivated.
+         */
         KOKKOS_INLINE_FUNCTION discrete_element_type
         eval_deriv(DSpan1D derivs, ddc::Coordinate<Tag> const& x) const;
 
+        /** @brief Evaluates non-zero B-splines values and \f$n\f$ derivatives at a given coordinate
+         *
+         * The values and derivatives are computed for every B-splines with support at the given coordinate x. There are only (degree+1)
+         * B-splines which are non-zero at any given point. It is these B-splines which are derivated.
+         * A spline approximation of a derivative at coordinate x is a linear
+         * combination of those B-splines derivatives weigthed with splines coefficients of the spline-transformed
```

Comment:
```suggestion
         * combination of those B-splines derivatives weighted with the spline coefficients of the spline-transformed
```

Reply:
"splines coefficients" still present

Reply:
https://github.com/CExA-project/ddc/pull/415#discussion_r1572057422

Reply:
Sorry I think I restored the Ss instead of removing ^^'